src/Libraries/CoreNodeModelsWpf/Controls/CustomSelectionControl.xaml.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Input;
+using System.Windows.Media;
+using CoreNodeModels.Input;
+
+namespace CoreNodeModelsWpf.Controls
+{
+    /// <summary>
+    /// Interaction logic for DropDownControl.xaml
+    /// </summary>
+    public partial class CustomSelectionControl : UserControl
+    {
+        /// <summary>
+        /// Create the control for the custom dropdown menu and editor
+        /// </summary>
+        public  CustomSelectionControl()
+        {
+            InitializeComponent();
+        }
+
+        private void EnumItemsListbox_PreviewKeyDown(object sender, KeyEventArgs e)
+        {
+            var listBox = sender as ListBox;
+            if (listBox == null)
+                return;
+
+            try
+            {
+                if (e.Key == Key.Tab)
+                {
+                    var textBoxes = new List<TextBox>();
+                    CollectChildrenOfType<TextBox>(listBox,textBoxes);
+
+                    for (int i = 0; i < textBoxes.Count - 1; i++)
+                    {
+                        if (textBoxes[i].IsKeyboardFocused)
+                        {
+                            textBoxes[i + 1].Focus();
+                            textBoxes[i + 1].CaretIndex = textBoxes[i + 1].Text.Length;
+                            return;
+                        }
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                var viewModel = DataContext as CustomSelectionNodeModel;
+                if (viewModel != null)
+                    viewModel.Log(ex);
+            }
+        }
+
+
+        private static void CollectChildrenOfType<T>(DependencyObject dependencyObj, List<T> children) where T : DependencyObject
```

Comment:
this helper already exists I believe:
https://github.com/DynamoDS/Dynamo/blob/master/src/DynamoCoreWpf/Utilities/WpfUtilities.cs#L62

please don't duplicate it if possible to avoid... but can you explain why you need to do this, it's going to be slow, and it's brittle if the xaml tree is changed by another dev in the future.